src/ably.js

```diff
@@ -0,0 +1,103 @@
+import * as Ably from 'ably';
+import { $store } from './store'
+import { ABLY_KEY } from "./config";
+
+// 1. RELOAD ( empty payload)
+// channel.publish("RELOAD", { });
+// 2. JUMP_TIME
+// channel.publish("JUMP_TIME", { timestamp: <EPOCH VAL> });
+// 3. SET_TEMPLATE
+// channel.publish("SET_TEMPLATE", { template: <TEMPLATE_FILE_FROM_TEMPLATE_DDL>});
+
+class AblyUpdates {
+    /**
+     * @param apiKey
+     */
+    constructor(apiKey) {
+        this._client = new Ably.Realtime(apiKey);
+        this._wsError = false;
+        this._worker = null;
+    }
+
+    run() {
+
+        if (this._worker) {
+            this._worker.terminate();
+            this._worker = null;
+        }
+
+        // connect handler
+        this._client.connection.on((stateChange) => {
+            const { current: state } = stateChange;
+            console.log(`AblyUpdates::connection WS ${state}`);
+            if (state === 'connected') {
+                this._wsError = false;
+                return;
+            }
+            if (state === 'suspended') {
+                if (!this._wsError) {
+                    this._wsError = true;
+                }
+                return;
+            }
+            if (state === 'failed') {
+                if (!this._wsError) {
+                    this._wsError = true;
+                }
+            }
+        });
+
+        $store.dispatch('getSummitId').then(summitId => {
+            $store.dispatch('getLocation').then(location => {
+                console.log(`AblyUpdates got summit ${summitId} and location ${location}`);
+
+                $store.dispatch('loadTemplate').then(({ data }) => {
+                    let initialTemplate = data[0]?.template;
+                    const currentPathname = window.location.pathname.replace('/', '');
+                    if (initialTemplate !== currentPathname) {
+                        let path = '/'
+                        let params = new URLSearchParams(window.location.href.split('?')[1]);
+                        let url = path + initialTemplate + '#/?' + params;
+                        window.location = url
+                    }
+                })
+
+                const channel = this._client.channels.get(`SIGNAGE:${summitId}:${location}`);
+
+                channel.subscribe('SET_TEMPLATE', (message) => {
+                    const { data: payload } = message;
+                    console.log("Received: " + JSON.stringify(payload));
+
+                    let { template } = payload;
```

Comment:
u should call method $store.setTemplate here
and have all this logic on that method